### PR TITLE
Cryptography using C

### PR DESCRIPTION
**Pull request by Rishabhpatel803**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/585"></a>

### CODE REVIEW

**Comment by Rishabhpatel803** [comment]
All files are checked and tested on various platforms.

**Comment by kvedala** [comment]
Thank you for the contributions 👍 
1. No need to close an existing PR, we can continue working on the comments made there. This will ensure for the reviewers to know exactly what has been reviewed and the progress along with it.
2. Please add appropriate documentation to the functions as in other files in the repo. See [this file](https://github.com/TheAlgorithms/C/blob/master/numerical_methods/newton_raphson_root.c) for an example on documenting the code for this repo. We are following the [Doxygen](http://doxygen.nl/manual/docblocks.html) standard for documentation.

This was mentioned in the PR description as one of the checklist for the authors. Please ensure the description is completed the checklist is followed.

Thank you

File: ciphers/caeser_cipher.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// Code by Rishabh Patel
```
Comment (Rishabhpatel803) [inline ciphers/caeser_cipher.c:1]: /**
 * @file
 * \brief Find approximate solution for \f$f(x) = 0\f$ using
 * Newton-Raphson interpolation algorithm.
 *
 * \author [Krishna Vedala](https://github.com/kvedala)
 */

**Comment by kvedala** [comment]
Thank you.
Can you please see the structure of the codes as below:
```c
/**
 * @file 
 * @brief Add one line description here
 * @details 
 * This is a multi line
 * description containing links, references,
 * math equations, etc
 * @author [Name](https://github.com/handle)
 * @see related_file.cpp, another_file.cpp
 */

#include 

/**
 * Structure documentation
 */
struct strct_name {
    int var1;  /**< short info of this variable */
    char *msg;  /**< short info */
};

/**
 * Function documentation 
 * @param param1 on-line info about param1
 * @param param2 on-line info about param2
 * @returns `true` if ...
 * @returns `false` if ... 
 */
bool func(int param1, T param2) {
    // function statements here 
    if(/*something bad*/)
        return false;

    return true;
}

/** Test function
   * @returns None
   */
void test() {
 /* some statements */
 assert(func(...) == ...); // this ensures that the algorithm works as expected 

 // can have multiple checks
}

/** Main function
  * @returns 0 on clean exit
  * @returns nonzero on error.
  */
int main(int argc, char *argv[]) {
    // code here
    return 0;
}
```

**Review by swapnanildutta — DISMISSED** [review]

**Comment by Rishabhpatel803** [comment]
Added documentation to the code

**Comment by Rishabhpatel803** [comment]
No need of test cases in code therefore not implemented test cases

**Comment by kvedala** [comment]
Thank you, @Rishabhpatel803 
There is definitely a need of test cases. If you see the other programs in the folder, you will see the following steps to verify that the code works:
1. Create one or more test string(s): "Hello World!"
2. Encrypt it to some text "gibberish"
3. decrypt this text back. By logic, the original string should be obtained verbatim. This is where you check using `strcmp` or similar and ensure using `assert` that the result is 0.

**Comment by Rishabhpatel803** [comment]
Ok i will update it by tomorrow

**Comment by Rishabhpatel803** [comment]
done adding test cases to the program

File: ciphers/playfair_cipher.c [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+ * In playfair cipher unlike traditional cipher we encrypt a pair of alphabets(digraphs) instead of a single alphabet
+ * \author [Rishabh Patel](https://github.com/Rishabhpatel803)
+ */
+
+#include <ctype.h>
+#include <stdio.h>
+#include <string.h>
+#define MX 5
+
+int choice;
```
Comment (kvedala) [inline ciphers/playfair_cipher.c:14]: Use of global variables is discouraged. Please pass variables as function arguments.

File: ciphers/playfair_cipher.c [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+/*
+ * designing the function of encryption and
+ * decryption using playfair cipher
+ */
+void playfair(char ch1, char ch2, char key[MX][MX]) {
```
Comment (kvedala) [inline ciphers/playfair_cipher.c:19]:
```suggestion
void playfair(char ch1, char ch2, char **key) {
```
Can the function be made better by allowing any size of keys and arrays? Not mandatory, but a suggestion.

File: ciphers/playfair_cipher.c [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+ * @brief: Playfair cipher was the first practical digraph substitution cipher
+ * @details: The scheme was invented in 1854 by Charles Wheatstone but was named after Lord Playfair who promoted the use of the cipher.
+ * In playfair cipher unlike traditional cipher we encrypt a pair of alphabets(digraphs) instead of a single alphabet
+ * \author [Rishabh Patel](https://github.com/Rishabhpatel803)
+ */
+
+#include <ctype.h>
+#include <stdio.h>
+#include <string.h>
+#define MX 5
```
Comment (kvedala) [inline ciphers/playfair_cipher.c:12]: If using macros, please use descriptive names of at least 5 characters long. Something like `KEY_SIZE`?

File: ciphers/playfair_cipher.c [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+            }
+        }
+    }
+}
+
+void test() 
+{ 
+    char msg[] = "Playfair", key[] = "Good";
+    choice = 1;
+    printf("First Test Case");
```
Comment (kvedala) [inline ciphers/playfair_cipher.c:188]: Please refer to the suggestion made for the teaser cipher code on implementing a self-test funcition.

**Pull request closed without merging**